pandas/tests/series/indexing/test_setitem.py

```diff
@@ -126,6 +126,14 @@ def test_setitem_boolean_different_order(self, string_series):
 
         tm.assert_series_equal(copy, expected)
 
+    def test_setitem_boolean_python_list(self):
+        # GH19406
+        ser = Series([None, "b", None])
+        mask = [0, 2]
```

Comment:
Actually I think the mask should be ``boolean``, something like ``[True, False, True]``. Could parametrize over list, np.array and pd.Series (Series is the case from the issue)